tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpUploadTestBase.java

```diff
@@ -355,6 +360,7 @@ private void testUploadMessages(
             msg.setAddress(getEndpointName());
             final Promise<?> sendingComplete = Promise.promise();
             final Handler<ProtonSender> sendMsgHandler = replenishedSender -> {
+                replenishedSender.setQoS(senderQoS);
```

Comment:
the QoS of the sender is being set when opening the sender link. Setting this property after the link has been established has no impact.

Reply:
While it might be true what you are saying, setting this property does have an impact:

If you comment this line the test will actually fail, because the forwarded application QoS property does not match the expected value.

In order to double check I inserted two debug log statements in _VertxBasedAmqpProtocolAdapter_ on line 1112 and 1116 printing out whether the message is sent as AT_LEAST_ONCE or as AT_MOST_ONCE.

When also removing the AT_LEAST_ONCE QoS in _TelemetryAmqpIT_'s _senderQoSTypes()_ method I observe that all messages are sent with AT_LEAST_ONCE QoS in the AMQP protocol adapter's log and also the test fails.

I ran the test with: _mvn verify -Prun-tests,jaeger,useRunningContainers -Ddocker.keepRunning -Dit.test=TelemetryAmqpIT#testUploadMessagesUsingSaslPlain_

Maybe this is a side-effect? Or is it a problem with my machine?

Reply:
Oh, you are working on the 1.3.x branch. In 1.3.x the AMQP adapter only supports opening link with *sender-settle-mode* `unsettled` (see https://github.com/eclipse/hono/blob/c63b042d2a79b7e8480e1995ce2db73175200159/tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterTestBase.java#L123). We are therefore not able to send *settled* messages from the test case. I guess it only makes sense to check for the correct QoS level in downstream messages for message being sent with AT_LEAST_ONCE semantics ...
For 1.5.0, we have changed the AMQP adapter to now also accept links with *snd-settle-mode* `settled`, your test should therefore work on master. However, I am afraid that it will also not work on 1.4.x